### PR TITLE
webgpu: fix non uniform flow control warning

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6443)

### CODE REVIEW

**Comment by xhcao** [comment]
Hi, @qjia7 @axinging Please take a look, thank you.

File: tfjs-backend-webgpu/src/shader_preprocessor.ts [hunk 1]
```diff
@@ -128,19 +129,19 @@ export function makeShader(
...
-          localId.y * workGroupSizeX + localId.x;
-      let workGroupID = (globalId - localId)/vec3<u32>(
-          workGroupSizeX, workGroupSizeY, workGroupSizeZ);
-
-      return i32((workGroupID.z * numWorkgroups.x * numWorkgroups.y +
-        workGroupID.y * numWorkgroups.x + workGroupID.x) *
-        (workGroupSizeX * workGroupSizeY * workGroupSizeZ) +
-        localInvocationIndex);
+      ${
+      program.dispatch[1] === 1 && program.dispatch[2] === 1 ?
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/shader_preprocessor.ts:133]: `program.dispatch[1] === 1 && program.dispatch[2] === 1` should be part of the shader key in `makeShaderKey`?

**Review by qjia7 — APPROVED** [review]
LGTM after fixing bots.

**Comment by xhcao** [comment]
@qjia7 Thanks.

**Review by qjia7 — APPROVED** [review]
Still LGTM, thanks.

**Review by gyagp — APPROVED** [review]
LGTM. We need to refactor ProgramParams in near future.

**Pull request merged**